src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,38 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private final StringBuilder result = new StringBuilder();
+    private final int dateIndex = 0;
+    private final int nameIndex = 1;
+    private final int workingHourIndex = 2;
+    private final int incomePerDayIndex = 3;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        int moneyEarned = 0;
+        result.delete(0,result.length());
```

Comment:
I assume you did this to pass the test which calls method two times? just make StringBuilder a local variable instead

Reply:
I don't want to create a new object when the method is called.